### DISCUSSION TITLE
Project class interface with DVC

### DISCUSSION THREAD

**Discussion by PythonFZ** [zincware/ZnTrack#19] [General]
What should the project class be able to do for us, regarding the DVC interface?

Currently the Project class supports 
- creating a DVC repository #49 
- running all stages #50 

Because we want users to be able to use MLSuite without knowing about GIT or DVC we need more methods to interface.
- show experiments and selected data from these experiements
- load experiments
- load files from multiple experiments e.g. https://github.com/zincware/PyTrack/blob/main/pytrack/interface/base.py
- querying multiple runs 

We should think about how we want to use `dvc exp run` `dvc repro` `dvc exp branch`, ... in MLSuite.

What functionality is crucial and currently missing and not listed above?
If generalizable most of this should be implemented in PyTrack not in MLSuite. But because PyTrack can not be used directly I created the discussion here.

**Comment by SamTov** [comment]
I think that the second point on if we want the users to not know git can be considered true. We should always as a base line consider that our users are new to computing and have little to no knowledge of git. They should be able to accomplish any task solely using MLSuite.

**Reply by PythonFZ** [reply]
`Because we want users to be able to use MLSuite without knowing about GIT or DVC we need more methods to interface.` Updated

**Comment by SamTov** [comment]
What we will need then is a wrapper for the DVC  git interface. So perhaps a bigger question is do you put this in PyTrack and have us call it from MLSuite?

**Reply by PythonFZ** [reply]
Moved the discussion to PyTrack. I guess this makes sense, so we can mention PRs and Commits

**Reply by SamTov** [reply]
Sounds good.
